tests/Doctrine/Tests/DBAL/Internal/CommitOrderCalculatorTest.php

```diff
@@ -0,0 +1,104 @@
+    {
+        $table1 = new Table('table1');
+        $table2 = new Table('table2');
+        $table3 = new Table('table3');
+        $table4 = new Table('table4');
+        $table5 = new Table('table5');
+
+        $this->calculator->addNode($table1->getName(), $table1);
+        $this->calculator->addNode($table2->getName(), $table2);
+        $this->calculator->addNode($table3->getName(), $table3);
+        $this->calculator->addNode($table4->getName(), $table4);
+        $this->calculator->addNode($table5->getName(), $table5);
+
+        $this->calculator->addDependency($table1->getName(), $table2->getName(), 1);
+        $this->calculator->addDependency($table2->getName(), $table3->getName(), 1);
+        $this->calculator->addDependency($table3->getName(), $table4->getName(), 1);
+        $this->calculator->addDependency($table5->getName(), $table1->getName(), 1);
+
+        $sorted = $this->calculator->sort();
+
+        // There is only 1 valid ordering for this constellation
+        $correctOrder = [$table5, $table1, $table2, $table3, $table4];
+
+        $this->assertSame($correctOrder, $sorted);
+    }
+
+    public function testCommitOrdering2()
+    {
+        $table1 = new Table('table1');
+        $table2 = new Table('table2');
+
+        $this->calculator->addNode($table1->getName(), $table1);
+        $this->calculator->addNode($table2->getName(), $table2);
+
+        $this->calculator->addDependency($table1->getName(), $table2->getName(), 0);
+        $this->calculator->addDependency($table2->getName(), $table1->getName(), 1);
+
+        $sorted = $this->calculator->sort();
+
+        // There is only 1 valid ordering for this constellation
+        $correctOrder = [$table2, $table1];
+
+        $this->assertSame($correctOrder, $sorted);
+    }
+
+    public function testCommitOrdering3()
+    {
+        // this test corresponds to the GH7259Test::testPersistFileBeforeVersion functional test
+        $table1 = new Table('table1');
+        $table2 = new Table('table2');
+        $table3 = new Table('table3');
+        $table4 = new Table('table4');
+
+        $this->calculator->addNode($table1->getName(), $table1);
+        $this->calculator->addNode($table2->getName(), $table2);
+        $this->calculator->addNode($table3->getName(), $table3);
+        $this->calculator->addNode($table4->getName(), $table4);
+
+        $this->calculator->addDependency($table4->getName(), $table1->getName(), 1);
+        $this->calculator->addDependency($table1->getName(), $table2->getName(), 1);
+        $this->calculator->addDependency($table4->getName(), $table3->getName(), 1);
+        $this->calculator->addDependency($table1->getName(), $table4->getName(), 0);
+
+        $sorted = $this->calculator->sort();
+
+        // There is only multiple valid ordering for this constellation, but
+        // the class4, class1, class2 ordering is important to break the cycle
+        // on the nullable link.
+        $correctOrders = [
```

Comment:
Does it mean that the order is not deterministic? Otherwise, we should either make a specific assertion or test all possible outcomes explicitly.

Reply:
I don't know, this is copied from the ORM.